context.go

```diff
@@ -119,19 +114,28 @@ func (ps *Set) Add(p *Plugin) error {
 	return nil
 }
 
-// Get returns the first plugin by its type
-func (ps *Set) Get(t Type) (interface{}, error) {
-	for _, v := range ps.byTypeAndID[t] {
-		return v.Instance()
+// Get returns the plugin with the given type and id
+func (ps *Set) Get(t Type, id string) *Plugin {
+	p, ok := ps.byTypeAndID[t]
+	if !ok {
+		return nil
 	}
-	return nil, fmt.Errorf("no plugins registered for %s: %w", t, ErrPluginNotFound)
+	return p[id]
 }
 
 // GetAll returns all initialized plugins
 func (ps *Set) GetAll() []*Plugin {
 	return ps.ordered
 }
 
+// Get returns the first plugin by its type
```

Comment:
Do we still want just the first plugin here?  I'm not quite following the situation where we'd only want just one, if there are multiple loaded plugins of the same type.